app/src/main/java/com/codesoom/assignment/config/AuthInterceptor.java

```diff
@@ -0,0 +1,49 @@
+package com.codesoom.assignment.config;
+
+import com.codesoom.assignment.application.AuthenticationService;
+import com.codesoom.assignment.errors.InvalidTokenException;
+import io.jsonwebtoken.JwtException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Slf4j
+@RequiredArgsConstructor
+public class AuthInterceptor implements HandlerInterceptor {
+
+    private final AuthenticationService authenticationService;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        String authorization = request.getHeader("Authorization");
+        log.info(">>>{}" + authorization);
+        if (authorization != null && authorization.startsWith("Bearer ")) {
+            log.info(">>>{}" + authorization.startsWith("Bearer "));
+            String token = authorization.substring("Bearer ".length());
+            log.info(">>>{}", token);
+            try {
+                authenticationService.parseToken(token);
+                return true;
+            } catch (InvalidTokenException e) {
+                throw new InvalidTokenException();
+            }
+
+        } else {
+            throw new InvalidTokenException();
+        }
+    }
```

Comment:
```suggestion
    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
        String authorization = request.getHeader("Authorization");
        log.info(">>>{}" + authorization);
        if (authorization == null || !authorization.startsWith("Bearer ")) {
            throw new InvalidTokenException();
        }
        
        log.info(">>>{}" + authorization.startsWith("Bearer "));
        String token = authorization.substring("Bearer ".length());
        log.info(">>>{}", token);
        try {
            authenticationService.parseToken(token);
            return true;
        } catch (InvalidTokenException e) {
            throw new InvalidTokenException();
        }
    }
```

이렇게 if중첩을 줄이면 좋을 것 같습니다